zetta_utils/builder/parser.py

```diff
@@ -18,7 +22,12 @@
 
 
 @typechecked
-def register(name: str, versions=None) -> Callable[[T], T]:
+def register(
+    name: str,
+    cast_to_vec3d: Optional[List[str]] = None,
+    cast_to_intvec3d: Optional[List[str]] = None,
```

Comment:
Is it a good idea to:
```python
default_converter={
   'resolution': Vec3D, 
   'bcube': BoundingCube.from_slices
}
``` 

How unlikely is it that we'd need to do this for new types? How worried are we about users abusing this?

Reply:
I think it's unlikely that we'd need to do this for new types, but I think it is likely that we will want to do this everywhere once we start doing it to avoid `@builder` syntax. Not worried about users abusing it per se, but I'm worried that we will forget to write it for some functions or update it when we change the function signature and tracking down what's failing where is going to be difficult. If we are going to add a default converter, I'd prefer that it be done algorithmically from the type annotations somehow, though I don't know how feasible that is.

Reply:
Ok, then how about we stick to a more generic syntax for now purely because it's fewer lines of code/fewer ifs/fewer tests:
```python
@builder.register(
    "foo", 
     default_converters={
           ....
     }
)
def foo(...)

Reply:
Also, how about the following behavior changes:
1. `Vec3D((x, y, z)) -> Vec3D(x, y, z)
2. Auto conversion behaves differently based on the given value. If it's a dict, it's assumed to be `kwargs`. If it's a list, it's assumed to be `args`. Otherwise, it's assumed to be a complete object and auto conversion is not applied.

Reply:
Sounds reasonable. Let me see if I can make it work.